### PR TITLE
feature(sage-icon) - add color prop to sage icon

### PR DESCRIPTION
**Pull request by ju-Skinner**
# Description

Adding the `color` attribute to allow customers to specify a color they would like the icon to be rendered. The attribute allows both string and hex formats

Fixes #(issue)
https://kajabi.atlassian.net/browse/DSS-283

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests you've added and run to verify your changes.
Provide instructions so that we can reproduce.
Please also list any relevant details for your test configuration.

- [x] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [x] tested manually
- [x] other: Storybook

**Test Configuration**:

- Sage versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

### CODE REVIEW

**Comment by ju-Skinner** [comment]
Please note that there is a dependency on PR #62. This PR CANNOT be merged until its dependency has been merged. Thank you.

File: libs/icons/src/components/sage-icon/sage-icon.tsx [hunk 1]
```diff
@@ -144,3 +159,13 @@ export class SageIcon {
...
 }
+
+const createColorClasses = (color: string | undefined) => {
+  return color
+   ? {
+       'sage-color': true,
+       [`sage-color-${color}`]: true,
+     }
+   : null;
+
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.tsx]: Delete this line

File: libs/icons/package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "test": "stencil test --spec --e2e",
     "test.spec": "stencil test --spec",
     "test.spec.watch": "stencil test --spec --watch",
+    "test.watch": "stencil test --spec --e2e --watch --coverage",
```
Comment (QuintonJason) [inline libs/icons/package.json:48]: 🏆 Nice add

File: libs/icons/src/components/sage-icon/sage-icon.scss [hunk 3]
```diff
@@ -1,20 +1,25 @@
 :host {
-  display: inline-block;
-
-  width: 1em;
-  height: 1em;
+  box-sizing: content-box;
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.scss]: I'm not sure if this is doing anything, does this need to be here?

File: libs/icons/src/components/sage-icon/sage-icon.scss [hunk 4]
```diff
@@ -1,20 +1,25 @@
...
 
   contain: strict;
 
+  display: inline-block;
+
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.scss]: Please remove the spaces between the rules here and above in this declaration

File: libs/icons/src/components/sage-icon/sage-icon.scss [hunk 5]
```diff
@@ -23,11 +28,11 @@ svg {
   width: 100%;
 }
 
-/* Icon Sizes
- * -----------------------------------------------------------
- */
+// Icon Sizes
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.scss]: Remove the white space on line 26

File: libs/icons/src/components/sage-icon/sage-icon.scss [hunk 6]
```diff
@@ -23,11 +28,11 @@ svg {
...
+// -----------------------------------------------------------
+//
 
- :host(.sage-icon-small) {
+:host(.sage-icon-small) {
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.scss]: The icon size looks like it's being determined by the inline `style`s, does this section still need to exist?

File: libs/icons/src/components/sage-icon/sage-icon.tsx [hunk 7]
```diff
@@ -38,6 +45,8 @@ export class SageIcon {
     | 'auto'
     | string = 'regular'
 
+
```
Comment (QuintonJason) [inline libs/icons/src/components/sage-icon/sage-icon.tsx]: Remove the extra space

**Review by QuintonJason — CHANGES_REQUESTED** [review]
Good stuff. Left a couple comments.

**Review by QuintonJason — APPROVED** [review]
Great work on this PR! Good to go

**Review by pixelflips — APPROVED** [review]
LGTM! 👍🏼

**Pull request merged**